### ISSUE TITLE
Segmentfilter saving issues

### ISSUE THREAD

**Issue by EsdersGmbH** [mautic/mautic#13879]
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.1.2

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When I try to add a new filter to the segment filter and I click save. There is a big chance I get an error message without any text and all fields are empty and saying something like "Please insert value".

All value are filled in and sended to the server, but the server response with this strange error as you can see in the picture.

![mautic](https://github.com/mautic/mautic/assets/59967049/1dd93b2e-f6e1-4609-a98a-4fc08fb5d78d)

### How can we reproduce this issue?

Step 1: Open a segment
Step 2: Add a new or more or filters with and filters applied
Step 3: Save

### Relevant log output

```shell
Form data sended to the server
leadlist[filters][15] and leadlist[filters][16] are the newly added filters.

leadlist[name]: My Segment
leadlist[alias]: my-segment
leadlist[publicName]: My Segment
leadlist[description]: 
leadlist[filters][0][glue]: and
leadlist[filters][0][operator]: in
leadlist[filters][0][properties][filter][]: 19
leadlist[filters][0][field]: leadlist
leadlist[filters][0][type]: leadlist
leadlist[filters][0][object]: lead
leadlist[filters][1][glue]: and
leadlist[filters][1][operator]: startsWith
leadlist[filters][1][properties][filter]: 79
leadlist[filters][1][field]: zipcode
leadlist[filters][1][type]: text
leadlist[filters][1][object]: lead
leadlist[filters][2][glue]: or
leadlist[filters][2][operator]: in
leadlist[filters][2][properties][filter][]: 19
leadlist[filters][2][field]: leadlist
leadlist[filters][2][type]: leadlist
leadlist[filters][2][object]: lead
leadlist[filters][3][glue]: and
leadlist[filters][3][operator]: startsWith
leadlist[filters][3][properties][filter]: 77
leadlist[filters][3][field]: zipcode
leadlist[filters][3][type]: text
leadlist[filters][3][object]: lead
leadlist[filters][4][glue]: or
leadlist[filters][4][operator]: in
leadlist[filters][4][properties][filter][]: 19
leadlist[filters][4][field]: leadlist
leadlist[filters][4][type]: leadlist
leadlist[filters][4][object]: lead
leadlist[filters][5][glue]: and
leadlist[filters][5][operator]: startsWith
leadlist[filters][5][properties][filter]: 78
leadlist[filters][5][field]: zipcode
leadlist[filters][5][type]: text
leadlist[filters][5][object]: lead
leadlist[filters][6][glue]: or
leadlist[filters][6][operator]: in
leadlist[filters][6][properties][filter][]: 19
leadlist[filters][6][field]: leadlist
leadlist[filters][6][type]: leadlist
leadlist[filters][6][object]: lead
leadlist[filters][7][glue]: and
leadlist[filters][7][operator]: startsWith
leadlist[filters][7][properties][filter]: 72
leadlist[filters][7][field]: zipcode
leadlist[filters][7][type]: text
leadlist[filters][7][object]: lead
leadlist[filters][8][glue]: and
leadlist[filters][8][operator]: =
leadlist[filters][8][properties][filter]: Germany
leadlist[filters][8][field]: country
leadlist[filters][8][type]: country
leadlist[filters][8][object]: lead
leadlist[filters][9][glue]: or
leadlist[filters][9][operator]: in
leadlist[filters][9][properties][filter][]: 19
leadlist[filters][9][field]: leadlist
leadlist[filters][9][type]: leadlist
leadlist[filters][9][object]: lead
leadlist[filters][10][glue]: and
leadlist[filters][10][operator]: startsWith
leadlist[filters][10][properties][filter]: 88
leadlist[filters][10][field]: zipcode
leadlist[filters][10][type]: text
leadlist[filters][10][object]: lead
leadlist[filters][11][glue]: and
leadlist[filters][11][operator]: =
leadlist[filters][11][properties][filter]: Germany
leadlist[filters][11][field]: country
leadlist[filters][11][type]: country
leadlist[filters][11][object]: lead
leadlist[filters][12][glue]: or
leadlist[filters][12][operator]: in
leadlist[filters][12][properties][filter][]: 19
leadlist[filters][12][field]: leadlist
leadlist[filters][12][type]: leadlist
leadlist[filters][12][object]: lead
leadlist[filters][13][glue]: and
leadlist[filters][13][operator]: startsWith
leadlist[filters][13][properties][filter]: 87
leadlist[filters][13][field]: zipcode
leadlist[filters][13][type]: text
leadlist[filters][13][object]: lead
leadlist[filters][14][glue]: and
leadlist[filters][14][operator]: =
leadlist[filters][14][properties][filter]: Germany
leadlist[filters][14][field]: companycountry
leadlist[filters][14][type]: country
leadlist[filters][14][object]: company
leadlist[filters][15][glue]: or
leadlist[filters][15][operator]: in
leadlist[filters][15][properties][filter][]: 19
leadlist[filters][15][field]: leadlist
leadlist[filters][15][type]: leadlist
leadlist[filters][15][object]: lead
leadlist[filters][16][glue]: and
leadlist[filters][16][operator]: =
leadlist[filters][16][properties][filter]: 123235
leadlist[filters][16][field]: companyzipcode
leadlist[filters][16][type]: text
leadlist[filters][16][object]: company
leadlist[category]: 14
leadlist[isGlobal]: 1
leadlist[isPreferenceCenter]: 0
leadlist[isPublished]: 1
leadlist[unlockModel]: lead.list
leadlist[unlockId]: 109
leadlist[_token]: TOKEN
leadlist[buttons][save]:
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Hi there, which version of Mautic are you on? Have you tried to both clear your cache, and rebuild your Mautic assets with the command `bin/console m:a:g`?

**Comment by EsdersGmbH** [comment]
Currently we are running Mautic 5.1.0. But your command seems to fix the issue. I will test it and then I will close the issue.

**Comment by patrykgruszka** [comment]
I can confirm this issue on current 5.1. Please test the fix #14222.

**Comment by RCheesley** [comment]
The fix is merged now, so I'm going to close this issue. Thanks @patrykgruszka 🚀
